### PR TITLE
Add Object permissions

### PR DESCRIPTION
**Pull request by cleder**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -11,7 +11,8 @@ In Django rest Framework the permission classes work as:
 This implementation of permission classes takes a list of functions that
 determine if the access is allowed. If **any** of the functions returns True,
```
Comment (amusikal) [inline README.rst]: Being super-pedantic:
'''If **any** of the functions return True''' - i.e. return is singular in this case.

'''[...], if **none** [...]''' - Redundant? Since any=pass.

File: drfdapc/test_permissions.py [hunk 2]
```diff
@@ -24,6 +32,95 @@ def tearDown(self):
...
+        self.user.is_staff = True
+        self.user.save()
+        request.user = self.user
+        self.assertTrue(permission.has_object_permission(request, None, obj))
+        self.user.is_superuser = True
+        self.user.is_staff = False
+        self.user.save()
+        request.user = self.user
+        self.assertFalse(permission.has_object_permission(request, None, obj))
+
```
Comment (amusikal) [inline drfdapc/test_permissions.py]: I know this a totally unimportant thing, but I'd always differentiate setup-plus-assertion blocks a little more. YMMV though :)

```
 +    def check_object_permission(self, permission, request):
 +        """
 +        Test the permission for a request for anonymous, staff and superuser.
 +
 +        Assuming that only staff can access the object when the object itself
 +        forbids access.
 +        """
 +        obj = mock.Mock()

 +        # allow access
 +        obj.allows_access = True
 +        request.user = AnonymousUser()
 +        self.assertTrue(permission.has_object_permission(request, None, obj))

 +        self.user.is_superuser = False
 +        self.user.is_staff = True
 +        self.user.save()
 +        request.user = self.user
 +        self.assertTrue(permission.has_object_permission(request, None, obj))

 +        self.user.is_superuser = True
 +        self.user.is_staff = False
 +        self.user.save()
 +        request.user = self.user
 +        self.assertTrue(permission.has_object_permission(request, None, obj))

 +        # restrict access
 +        obj.allows_access = False
 +        request.user = AnonymousUser()
 +        self.assertFalse(permission.has_object_permission(request, None, obj))

 +        self.user.is_superuser = False
 +        self.user.is_staff = True
 +        self.user.save()
 +        request.user = self.user
 +        self.assertTrue(permission.has_object_permission(request, None, obj))

 +        self.user.is_superuser = True
 +        self.user.is_staff = False
 +        self.user.save()
 +        request.user = self.user
 +        self.assertFalse(permission.has_object_permission(request, None, obj))
```

**Comment by amusikal** [comment]
Looks good to me. Great coverage :)
Feel free to ignore my pedantic comments... everything seems fine.

File: README.rst (hunk 1 above)
Comment (cleder) [inline README.rst]: Yes this is redundant I tend to repeat myself. It is to be absolutely clear to even the greatest Noob what this does.

**Pull request merged**